LibreNMS/Device/YamlDiscovery.php

```diff
@@ -211,6 +212,19 @@ public static function getValueFromData($name, $index, $discovery_data, $pre_cac
             $name = $discovery_data[$name];
         }
 
+        //create the sub-index values in order to try to match them with precache
+        $sub_indexes = explode('.', $index);
+        // parse sub_index options name with trailing colon and index
+        $sub_index = 0;
+        if (preg_match('/^(?<name>[^:]+(::[^:]+)?):(?<prefix>[^:-]*?)(?<index>\d+)(?:-(?<end>\d+))?(?<suffix>.*)$/', $name, $sub_index_matches)) {
```

Comment:
tiny bit more consistent 
```suggestion
        if (preg_match('/^(?<name>[^:]+(?:::[^:]+)?):(?<prefix>[^:-]*?)(?<index>\d+)(?:-(?<end>\d+))?(?<suffix>.*)$/', $name, $sub_index_matches)) {
```